File: src/main/java-graphql/io/github/graphql/model/ReactionResponseProjection.java
package io.github.graphql.model;

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField;
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection;

/**
 * Response projection for Reaction
 */
@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-06T19:47:47+0800"
)
public class ReactionResponseProjection extends GraphQLResponseProjection {

    public ReactionResponseProjection() {
    }

    @Override
    public ReactionResponseProjection all$() {
        return all$(3);
    }

    @Override
    public ReactionResponseProjection all$(int maxDepth) {
        this.content();
        this.createdAt();
        this.databaseId();
        this.id();
        if (projectionDepthOnFields.getOrDefault("ReactionResponseProjection.ReactableResponseProjection.reactable", 0) <= maxDepth) {
            projectionDepthOnFields.put("ReactionResponseProjection.ReactableResponseProjection.reactable", projectionDepthOnFields.getOrDefault("ReactionResponseProjection.ReactableResponseProjection.reactable", 0) + 1);
            this.reactable(new ReactableResponseProjection().all$(maxDepth - projectionDepthOnFields.getOrDefault("ReactionResponseProjection.ReactableResponseProjection.reactable", 0)));
        }
        if (projectionDepthOnFields.getOrDefault("ReactionResponseProjection.UserResponseProjection.user", 0) <= maxDepth) {
            projectionDepthOnFields.put("ReactionResponseProjection.UserResponseProjection.user", projectionDepthOnFields.getOrDefault("ReactionResponseProjection.UserResponseProjection.user", 0) + 1);
            this.user(new UserResponseProjection().all$(maxDepth - projectionDepthOnFields.getOrDefault("ReactionResponseProjection.UserResponseProjection.user", 0)));
        }
        this.typename();
        return this;
    }

    public ReactionResponseProjection content() {
        return content(null);
    }

    public ReactionResponseProjection content(String alias) {
        fields.add(new GraphQLResponseField("content").alias(alias));
        return this;
    }

    public ReactionResponseProjection createdAt() {
        return createdAt(null);
    }

    public ReactionResponseProjection createdAt(String alias) {
        fields.add(new GraphQLResponseField("createdAt").alias(alias));
        return this;
    }

    public ReactionResponseProjection databaseId() {
        return databaseId(null);
    }

    public ReactionResponseProjection databaseId(String alias) {
        fields.add(new GraphQLResponseField("databaseId").alias(alias));
        return this;
    }

    public ReactionResponseProjection id() {
        return id(null);
    }

    public ReactionResponseProjection id(String alias) {
        fields.add(new GraphQLResponseField("id").alias(alias));
        return this;
    }

    public ReactionResponseProjection reactable(ReactableResponseProjection subProjection) {
        return reactable(null, subProjection);
    }

    public ReactionResponseProjection reactable(String alias, ReactableResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("reactable").alias(alias).projection(subProjection));
        return this;
    }

    public ReactionResponseProjection user(UserResponseProjection subProjection) {
        return user(null, subProjection);
    }

    public ReactionResponseProjection user(String alias, UserResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("user").alias(alias).projection(subProjection));
        return this;
    }

    public ReactionResponseProjection typename() {
        return typename(null);
    }

    public ReactionResponseProjection typename(String alias) {
        fields.add(new GraphQLResponseField("__typename").alias(alias));
        return this;
    }


}

File: src/main/kotlin-graphql/io/github/graphql/kotlin/model/ReactionResponseProjection.kt
package io.github.graphql.kotlin.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection
import java.util.Objects

/**
 * Response projection for Reaction
 */
@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-07T19:48:13+0800"
)
open class ReactionResponseProjection : GraphQLResponseProjection() {

    override fun `all$`(): ReactionResponseProjection = `all$`(3)

    override fun `all$`(maxDepth: Int): ReactionResponseProjection {
        this.content()
        this.createdAt()
        this.databaseId()
        this.id()
        if (projectionDepthOnFields.getOrDefault("ReactionResponseProjection.ReactableResponseProjection.reactable", 0) <= maxDepth) {
            projectionDepthOnFields["ReactionResponseProjection.ReactableResponseProjection.reactable"] = projectionDepthOnFields.getOrDefault("ReactionResponseProjection.ReactableResponseProjection.reactable", 0) + 1
            this.reactable(ReactableResponseProjection().`all$`(maxDepth - projectionDepthOnFields.getOrDefault("ReactionResponseProjection.ReactableResponseProjection.reactable", 0)))
        }
        if (projectionDepthOnFields.getOrDefault("ReactionResponseProjection.UserResponseProjection.user", 0) <= maxDepth) {
            projectionDepthOnFields["ReactionResponseProjection.UserResponseProjection.user"] = projectionDepthOnFields.getOrDefault("ReactionResponseProjection.UserResponseProjection.user", 0) + 1
            this.user(UserResponseProjection().`all$`(maxDepth - projectionDepthOnFields.getOrDefault("ReactionResponseProjection.UserResponseProjection.user", 0)))
        }
        this.typename()
        return this
    }

    fun content(): ReactionResponseProjection = content(null)

    fun content(alias: String?): ReactionResponseProjection {
        fields.add(GraphQLResponseField("content").alias(alias))
        return this
    }

    fun createdAt(): ReactionResponseProjection = createdAt(null)

    fun createdAt(alias: String?): ReactionResponseProjection {
        fields.add(GraphQLResponseField("createdAt").alias(alias))
        return this
    }

    fun databaseId(): ReactionResponseProjection = databaseId(null)

    fun databaseId(alias: String?): ReactionResponseProjection {
        fields.add(GraphQLResponseField("databaseId").alias(alias))
        return this
    }

    fun id(): ReactionResponseProjection = id(null)

    fun id(alias: String?): ReactionResponseProjection {
        fields.add(GraphQLResponseField("id").alias(alias))
        return this
    }

    fun reactable(subProjection: ReactableResponseProjection): ReactionResponseProjection = reactable(null, subProjection)

    fun reactable(alias: String?, subProjection: ReactableResponseProjection): ReactionResponseProjection {
        fields.add(GraphQLResponseField("reactable").alias(alias).projection(subProjection))
        return this
    }

    fun user(subProjection: UserResponseProjection): ReactionResponseProjection = user(null, subProjection)

    fun user(alias: String?, subProjection: UserResponseProjection): ReactionResponseProjection {
        fields.add(GraphQLResponseField("user").alias(alias).projection(subProjection))
        return this
    }

    fun typename(): ReactionResponseProjection = typename(null)

    fun typename(alias: String?): ReactionResponseProjection {
        fields.add(GraphQLResponseField("__typename").alias(alias))
        return this
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val that = other as ReactionResponseProjection
        return Objects.equals(fields, that.fields)
    }

    override fun hashCode(): Int = Objects.hash(fields)

}
